pipeline/review/main.go

```diff
@@ -76,60 +55,102 @@ func main() {
+	flagCommitReviewStatusTable := flag.String("commit-review-status-table", "",
+		"Name of the commit review status table. The value provided must be a fully qualified BigQuery table name of the form <project>:<dataset>.<table>")
+	flagIssuesTable := flag.String("issues-table", "",
+		"name of the issues table. The value provided must be a fully qualified BigQuery table name of the form <project>:<dataset>.<table>")
+
+	// Parse flags.
 	flag.Parse()
-	// initialize beam. This must be done after flag.Parse() but before any flag validation.
+
+	// Initialize beam. This must be done after flag.Parse(), but before any flag
+	// validation.
 	// https://cloud.google.com/dataflow/docs/guides/setting-pipeline-options#CreatePipelineFromArgs
 	beam.Init()
-	// parse commandline arguments into config
-	pipelineConfig, err := getConfigFromFlags(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to construct pipeline config: %w", err)
+
+	// Validate inputs.
+	var merr error
+	if *flagGitHubToken == "" && *flagGitHubAppID == "" {
+		merr = errors.Join(merr, fmt.Errorf("one of github token or github app id are required"))
 	}
-	// construct and execute the pipeline
-	pipeline := review.NewCommitApprovalPipeline(pipelineConfig)
-	if err := beamx.Run(ctx, pipeline); err != nil {
-		return fmt.Errorf("failed to execute pipeline: %w", err)
+	if *flagGitHubToken != "" && *flagGitHubAppID != "" {
+		merr = errors.Join(merr, fmt.Errorf("only one of github token or github app id are allowed"))
 	}
-	return nil
-}
 
-// getConfigFromFlags returns review.CommitApprovalPipelineConfig struct
-// using flag values. returns an error if any of the flags have malformed
-// data.
-func getConfigFromFlags(ctx context.Context) (*review.CommitApprovalPipelineConfig, error) {
-	githubTokenSupplier, err := getGitHubTokenSupplier(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to construct token supplier: %w", err)
-	}
-	token, err := githubTokenSupplier.GitHubToken(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get GitHub token: %w", err)
+	// Create qualified table names.
+	pushEventsTable, err := newQualifiedTableName(*flagPushEventsTable)
+	merr = errors.Join(merr, err)
+	commitReviewStatusTable, err := newQualifiedTableName(*flagCommitReviewStatusTable)
+	merr = errors.Join(merr, err)
+	issuesTable, err := newQualifiedTableName(*flagIssuesTable)
+	merr = errors.Join(merr, err)
+
+	// Create the GitHub token source.
+	var githubTokenSource githubauth.TokenSource
+	if *flagGitHubToken != "" {
+		githubTokenSource, err = githubauth.NewStaticTokenSource(*flagGitHubToken)
+		if err != nil {
+			merr = errors.Join(merr, fmt.Errorf("failed to create github static token source: %w", err))
+			return merr
+		}
+	} else {
+		sm, err := secretmanager.NewClient(ctx)
+		if err != nil {
+			merr = errors.Join(merr, fmt.Errorf("failed to create secret manager client: %w", err))
+			return merr
```

Comment:
nit. Can save a few more lines throughout
```suggestion
		if err != nil {
			return errors.Join(merr, fmt.Errorf("failed to create github static token source: %w", err))
		}
	} else {
		sm, err := secretmanager.NewClient(ctx)
		if err != nil {
			return errors.Join(merr, fmt.Errorf("failed to create secret manager client: %w", err))
```